### PR TITLE
Pipelining followups

### PR DESCRIPTION
**Pull request by tzaffi**
See: https://github.com/algorand/conduit/pull/147 for the continuation of this work

### CODE REVIEW

File: conduit/pipeline/pipeline.go [hunk 1]
```diff
@@ -696,8 +695,6 @@ func (p *pipelineImpl) Start() {
 			}
 		}
 	}(p.pipelineMetadata.NextRound)
-
-	<-p.ctx.Done()
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:700]: This is the health endpoint bugfix.

File: conduit/pipeline/common.go [hunk 2]
```diff
@@ -74,9 +76,19 @@ func Retries[X pluginInput, Y pluginOutput](f func(x X) (Y, error), x X, p *pipe
 	return
 }
 
-// RetriesNoOutput applies the same logic as Retries, but for functions that return no output.
-func RetriesNoOutput[X pluginInput](f func(x X) error, a X, p *pipelineImpl, msg string) (time.Duration, error) {
-	_, d, err := Retries(func(x X) (empty, error) {
+// TODO: probly the following function and its unit test should be axed
```
Comment (tzaffi) [inline conduit/pipeline/common.go:79]: ^^^^ !!!

File: conduit/plugins/importers/noop/sample.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name: noop
+# noop has no config
+config:
```
Comment (tzaffi) [inline conduit/plugins/importers/noop/sample.yaml]:
```suggestion
config:

```

File: pkg/cli/cli_test.go [hunk 4]
```diff
@@ -135,3 +140,46 @@ func TestLogFile(t *testing.T) {
...
+func TestHealthEndpoint(t *testing.T) {
+	healthPort := 7777
+	healthNet := fmt.Sprintf("http://localhost:%d/health", healthPort)
+
+	test := func(t *testing.T, address string) {
+		// Capture stdout.
+		stdout := os.Stdout
+		defer func() {
+			os.Stdout = stdout
+		}()
```
Comment (tzaffi) [inline pkg/cli/cli_test.go]:
```suggestion
```

File: pkg/cli/cli_test.go [hunk 5]
```diff
@@ -135,3 +140,41 @@ func TestLogFile(t *testing.T) {
 		require.Contains(t, dataStr, "\nWriting logs to file:")
 	})
 }
+
+func TestHealthEndpoint(t *testing.T) {
```
Comment (algochoi) [inline pkg/cli/cli_test.go:144]: nice test 👍

File: conduit/pipeline/common.go (hunk 2 above)
Comment (algochoi) [reply conduit/pipeline/common.go:79]: I agree that we probably don't need a no-input function right now -- is this+tests+above interface going to be reverted?

**Comment by tzaffi** [comment]
closing this PR and will re-open a new one shortly against `upstream/master`

**Pull request closed without merging**